tools/crm_mon_print.c

```diff
@@ -54,6 +54,35 @@ static int print_node_attributes(pcmk__output_t *out, pe_working_set_t *data_set
 static int print_failed_actions(pcmk__output_t *out, pe_working_set_t *data_set,
                                 gboolean print_spacer);
 
+static GListPtr
+build_uname_list(pe_working_set_t *data_set, const char *s) {
+    GListPtr unames = NULL;
+
+    if (s == NULL) {
+        /* Nothing was given so return a list of all node names. */
+        unames = g_list_append(unames, strdup("*"));
+    } else {
+        pe_node_t *node = pe_find_node(data_set->nodes, s);
+
+        if (node) {
+            /* The given string was a valid uname for a node.  Return a
+             * singleton list containing just that uname.
+             */
+            unames = g_list_append(unames, strdup(s));
```

Comment:
Always favor g_list_prepend() when possible, it's more efficient

Reply:
Done.